src/main/scala/mesosphere/marathon/core/task/update/impl/TaskStatusUpdateProcessorImpl.scala

```diff
@@ -43,13 +47,19 @@ class TaskStatusUpdateProcessorImpl @Inject() (
 
     val now = clock.now()
     val taskId = Task.Id(status.getTaskId)
+    val marathonTaskStatus = MarathonTaskStatus(status)
 
     taskTracker.task(taskId).flatMap {
       case Some(task) =>
         val taskStateOp = TaskStateOp.MesosUpdate(task, status, now)
         stateOpProcessor.process(taskStateOp).flatMap(_ => acknowledge(status))
 
-      case None if killWhenUnknown(status) =>
+      case None if terminalUnknown(marathonTaskStatus) =>
+        log.warn("Received terminal status update for unknown {}", taskId)
+        eventStream.publish(UnknownTaskTerminated(taskId, taskId.runSpecId, marathonTaskStatus))
+        acknowledge(status)
+
+      case None if killWhenUnknown(marathonTaskStatus) =>
```

Comment:
I am not sure how this behavior changes with Mesos 1.1 or future versions; until now, when killing a task that was scheduled on an agent that is currently not registered, Mesos sent a statusUpdate declaring the task is LOST (which makes sense because the master cannot forward the request to the agent).